plugins/registry.js

```diff
@@ -460,4 +460,10 @@ export default class PluginRegistry {
 
         return {id, showRHSPlugin: showRHSPlugin(id)};
     }
+
+    // Register a component to replace latex code with a component provided by the plugin.
```

Comment:
A high-level comment before diving in: I'm wondering if we might want to generalize this to support overriding arbitrary code blocks. Perhaps something like:
```
registerCodeBlockComponent(component, languages)
```

Where `languages` is an array -- in this case, `['tex', 'latex']. Then the component should expect to receive as a prop the content of the code block (perhaps `data-code` vs. forcing `data-latex`).

We could initially document this as only supporting the above two languages, but then in another effort, add support for arbitrary replacements without having to break the interface.

Reply:
@lieut-data this is definitely a good idea :+1: